src/libsyncengine/update_detection/update_detector/updatetreeworker.cpp

```diff
@@ -1181,7 +1181,20 @@ ExitCode UpdateTreeWorker::updateNodeWithDb(const std::shared_ptr<Node> parentNo
         if (node->hasChangeEvent(OperationTypeMove)) {
             // update parentDbId
             node->setMoveOriginParentDbId(dbNode.parentNodeId());
-            // TODO : should we set moveOrigin too?
+
+            SyncPath localPath;
+            SyncPath remotePath;
+            _syncDb->path(node->idb().value(), localPath, remotePath, found);
+
+            if (!_syncDb->path(node->idb().value(), localPath, remotePath, found)) {
+                LOG_SYNCPAL_WARN(_logger, "Error in SyncDb::path");
+                return ExitCodeDbError;
+            }
+            if (!found) {
+                LOG_SYNCPAL_WARN(_logger, "Failed to retrieve node for DB ID=" << node->idb().value());
+                return ExitCodeDataError;
+            }
+            node->setMoveOrigin(_side == ReplicaSideLocal ? localPath : remotePath);    // TODO : no need to keep both remote and local path since we do not rename the file locally any more
```

Comment:
```suggestion
            node->setMoveOrigin(_side == ReplicaSideLocal ? localPath : remotePath);    // TODO : no need to keep both remote and local paths since we do not rename the file locally anymore.
```